nixos/modules/services/misc/klipper.nix

```
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.klipper;
+  package = pkgs.klipper;
+  format = formats.ini;
+in
+{
+  ##### interface
+  options = {
+    services.klipper = {
+      enable = mkEnableOption "Klipper, the 3D printer firmware";
+
+      settings = mkOption {
+        type = format.type;
+        default = {};
+        description = ''
+          Configuration for Klipper. See <documentation xlink:href="https://www.klipper3d.org/Overview.html#configuration-and-tuning-guides"/>
+          for supported values.
+        '';
+      };
+
+      logDir = mkOption {
+        type = types.path;
```

I think this should be `types.str` so that the log dir isn't imported into the store.

```suggestion
        type = types.str;
```

---

Applied by hand